library/std/src/io/mod.rs

```diff
@@ -917,6 +917,17 @@ pub trait Read {
     /// # }
     /// ```
     ///
+    /// # Usage Notes
+    ///
+    /// `read_to_end` attempts to read a source until EOF, but many sources are continuous streams
+    /// that do not send EOF. In these cases, `read_to_end` will block indefinitely. Standard input
+    /// is one such stream which may be finite if piped, but is typically continuous. For example,
+    /// `cat <file> | <my_rust_program>` will correctly terminate with an `EOF` upon closure of cat.
+    /// Reading user input or running programs that remain open indefinitely will never terminate
+    /// the stream with `EOF` (e.g. `yes "Data" | pv | <my_rust_program`).
```

Comment:
Two tiny requests:

* `cat <file> | <my_rust_program` -> `cat file | my-rust-program`, since those are totally valid binary names they don't need the `<>`. (kebab case is slightly more common for binaries)
* `yes | my-rust-program` same as above, and I don't think `Data` or `pv` are relevant to the output here are they?

Reply:
pv isn't necessary in this example. The main point was to have a program loop indefinitely while outputting data to stdout. Data is a string here that will be printed to stdout. yes on its own will just print y to stdout. I will remove Data since it doesn't make any difference and it's probably more concise without it.

Hmm, I see you are right. There are more tools named with kebab case than snake case for binary names. I learned something new today and I have used Unix systems for 10 years...

Am I dating myself by dropping snake case on my non-Rust projects? Must be a habit from my Windows days.

![image](https://github.com/user-attachments/assets/812d7592-54b0-486a-bfdb-06a3bfd9f5f8)